client/src/components/app/App.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { Provider } from "react-redux";
+import { Redirect, Route, Switch } from "react-router-dom";
+
+import AllArticlesContainer from "components/article/AllArticlesContainer";
+import DetailedArticle from "components/article/DetailedArticle";
+import Login from "components/user/Login";
+import Logout from "components/user/Logout";
+import NoMatch from "./NoMatch";
+import Signup from "components/user/Signup";
+import TopBar from "components/app/TopBar";
+import User from "components/user/User";
+
+import store from "redux/store";
+import LocalStorage from "LocalStorage";
+
+class App extends React.Component {
+  state = {
+    loggedInUser: {},
+  };
+
+  constructor(props) {
+    super(props);
+    LocalStorage.subscribe(this.localStorageUpdated);
+  }
+
+  componentDidMount() {
+    this.localStorageUpdated();
+  }
+
+  localStorageUpdated = () => {
+    const loggedInUser = LocalStorage.getLoggedInUser();
+    this.setState({ loggedInUser });
+  };
+
+  isLoggedIn = () => {
+    return Object.keys(this.state.loggedInUser).length !== 0;
+  };
+
+  render() {
+    return (
+      <div>
+        <Route
+          path='/'
+          render={(props) => <TopBar {...props} loggedIn={this.isLoggedIn()} />}
+        />
+        <Switch>
```

Comment:
Can we move all routes to another file and import this file.

Reply:
Done